eppo_client/client.py

```diff
@@ -55,11 +53,15 @@ def get_string_assignment(
         except Exception as e:
             if self.__is_graceful_mode:
                 logger.error("[Eppo SDK] Error getting assignment: " + str(e))
-                return None
+                return default
             raise e
 
     def get_numeric_assignment(
-        self, subject_key: str, flag_key: str, subject_attributes=dict()
+        self,
+        subject_key: str,
+        flag_key: str,
+        subject_attributes=dict(),
```

Comment:
(Same for all `get_<type>_assignment` functions.) we should avoid this pattern, as it will break if we change subject_attributes by adding the subject key to it. (It's a [common Python gotcha](https://web.archive.org/web/20200221224620id_/http://effbot.org/zone/default-values.htm).)